### PR TITLE
Add napari.app v0.4.2

### PR DESCRIPTION
**Pull request by tlambert03**
Please consider this new cask for napari, a fast multidimensional image viewer written in python:
https://github.com/napari/napari
https://napari.org

thanks!

Verified:

- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.  
  _( I was unable to run this locally, but it has passed on CI)_
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions)

**I'm adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew install --cask {{cask_file}}` worked successfully.
- [x] `brew uninstall --cask {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

**Comment by tlambert03** [comment]
one more note: I see that a `zap` is usually requested.  In this case, there are no settings/preferences saved by the app outside of the .app bundle (yet).  In the future, we plan to look for user-editable config files in `~/.config/napari/` ... but since these are both directly user-editable and program editable, I'm not sure that they fall under the [zap stanza purpose](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md#zap-stanza-purpose) ... but let me know if I should add it.

File: Casks/napari.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cask "napari" do
+  version "0.4.2"
+  sha256 "8a0481bf710fd2225a7777dbc94343caaa03928940d66e83b80d326ea61650ae"
+
+  # github.com/napari/napari was verified as official when first introduced to the cask
+  url "https://github.com/napari/napari/releases/download/v#{version}/napari-#{version}-macOS.zip"
+  appcast "https://github.com/napari/napari/releases.atom"
+  name "napari"
+  desc "Fast, interactive, multi-dimensional image viewer for Python"
```
Comment (vitorgalvao) [inline Casks/napari.rb:9]: “Fast, interactive, multi-dimensional” is marketing fluff. Just `Image viewer for Python`.

File: Casks/napari.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  homepage "https://napari.org/"
+
+  depends_on macos: "<= :catalina"
```
Comment (vitorgalvao) [inline Casks/napari.rb:12]: Why? Will it ever support Big Sur?

File: Casks/napari.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+cask "napari" do
+  version "0.4.2"
+  sha256 "8a0481bf710fd2225a7777dbc94343caaa03928940d66e83b80d326ea61650ae"
+
+  # github.com/napari/napari was verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/napari.rb:5]: Need a trailing slash on `github.com/napari/napari`

File: Casks/napari.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  url "https://github.com/napari/napari/releases/download/v#{version}/napari-#{version}-macOS.zip"
+  appcast "https://github.com/napari/napari/releases.atom"
+  name "napari"
+  desc "Fast, interactive, multi-dimensional image viewer for Python"
+  homepage "https://napari.org/"
+
+  depends_on macos: "<= :catalina"
+  container nested: "napari-#{version}.dmg"
+
+  app "napari.app"
```
Comment (vitorgalvao) [inline Casks/napari.rb:15]: Yes, please add the `zap`.

**Review by vitorgalvao — DISMISSED** [review]
Welcome. In the future please don’t untick the box that allows edits from maintainers. It makes our work considerably slower and harder with unnecessary back and forth.

File: Casks/napari.rb (hunk 1 above)
Comment (tlambert03) [reply Casks/napari.rb:9]: > Welcome. In the future please don’t untick the box that allows edits from maintainers. It makes our work considerably slower and harder with unnecessary back and forth.

Thanks for the review!  Definitely didn't untick that box... would be more than happy to have you edit it directly.  Not sure why it's disabled ... (maybe creating the PR view the `gh` cli?).  Sorry, I don't see it available on the right to re-enable either.

File: Casks/napari.rb (hunk 2 above)
Comment (tlambert03) [reply Casks/napari.rb:12]: Yep, probably the next patch release (0.4.3), napari itself _does_ support big sur, but the openGL dependencies require python 3.9, and one of our dependencies needs to release a patch update to support 3.9, so we are still shipping the bundle with py3.8 for now.  But will remove the Catalina pin as soon as we can reliably support it.

File: Casks/napari.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/napari.rb:9]: > Not sure why it's disabled ... (maybe creating the PR view the `gh` cli?).

Probably that, yes.

> Sorry, I don't see it available on the right to re-enable either.

Don’t worry about it for now, if you can do the requested fixes.

File: Casks/napari.rb (hunk 4 above)
Comment (tlambert03) [reply Casks/napari.rb:15]: will do!

File: Casks/napari.rb (hunk 2 above)
Comment (vitorgalvao) [reply Casks/napari.rb:12]: Sound good.

File: Casks/napari.rb (hunk 1 above)
Comment (tlambert03) [reply Casks/napari.rb:9]: > “Fast, interactive, multi-dimensional” is marketing fluff.

absolutely, that was just the standard description from the app.  Mind if I leave in the "multi-dimensional" part?  that's kinda the main differentiating point of the app compared to standard image viewers.

**Comment by tlambert03** [comment]
changes made, thanks for your time.

**Comment by vitorgalvao** [comment]
It’s failing the `style` check ([issues are pointed out](https://github.com/Homebrew/homebrew-cask/pull/93839/files)).

**Comment by tlambert03** [comment]
whoops.  fixed now, thanks

**Comment by miccal** [comment]
Thank you for your contribution to `homebrew-cask` @tlambert03.

**Review by miccal — APPROVED** [review]

**Pull request merged**